pkg/runtime/reconciler.go

```diff
@@ -552,17 +616,31 @@ func (r *resourceReconciler) getRegion(
 	return ackv1alpha1.AWSRegion(r.cfg.Region)
 }
 
-// NewReconciler returns a new reconciler object that
+// NewReconciler returns a new reconciler object
 func NewReconciler(
 	sc acktypes.ServiceController,
 	rmf acktypes.AWSResourceManagerFactory,
 	log logr.Logger,
 	cfg ackcfg.Config,
 	metrics *ackmetrics.Metrics,
+) acktypes.AWSResourceReconciler {
+	return NewReconcilerWithClient(sc, nil, rmf, log, cfg, metrics)
+}
+
+// NewReconcilerWithClient returns a new reconciler object
```

Comment:
Instead of a `NewReconcilerWithClient` function, how about just adding a `WithClient` method to the `resourceReconciler` struct:

```go
// WithClient returns the reconciler after setting its internal k8s client.
// NOTE: Used in testing
func (r *resourceReconciler) WithClient(kc client.Client) *resourceReconciler {
	r.kc = kc
	return r
}
```